diff --git a/include/plat/bcm2837/plat/machine/devices.h b/include/plat/bcm2837/plat/machine/devices.h
index 96f07c2..7310ebd 100644
--- a/include/plat/bcm2837/plat/machine/devices.h
+++ b/include/plat/bcm2837/plat/machine/devices.h
@@ -18,6 +18,11 @@
 #define PADDR_OFFSET                0x3F000000
 
 #define INTC_BUSADDR                0x7E00B000
+#define VC_MBOX_BUSADDR             0x7E00B880
+#define PM_BUSADDR                  0x7E100000
+#define RNG_BUSADDR                 0x7E104000
+#define GPIO_BUSADDR                0x7E200000
+#define UART0_BUSADDR               0x7E201000
 #define UART_BUSADDR                0x7E215000
 #define SDHC_BUSADDR                0x7E300000
 #define USB2_BUSADDR                0x7E980000
@@ -29,6 +34,11 @@
 /* We convert from the VC CPU BUS addresses to ARM Physical addresses due to the extra
     VC (Video controller) MMU */
 #define INTC_PADDR                  (INTC_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
+#define VC_MBOX_PADDR               (VC_MBOX_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
+#define PM_PADDR                    (PM_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
+#define RNG_PADDR                   (RNG_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
+#define GPIO_PADDR                  (GPIO_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
+#define UART0_PADDR                 (UART0_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
 #define UART_PADDR                  (UART_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
 #define SDHC_PADDR                  (SDHC_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
 #define USB2_PADDR                  (USB2_BUSADDR-BUS_ADDR_OFFSET+PADDR_OFFSET)
diff --git a/include/plat/bcm2837/plat/machine/hardware.h b/include/plat/bcm2837/plat/machine/hardware.h
index 1eb168b..f1d330e 100644
--- a/include/plat/bcm2837/plat/machine/hardware.h
+++ b/include/plat/bcm2837/plat/machine/hardware.h
@@ -56,6 +56,11 @@ const p_region_t BOOT_RODATA avail_p_regs[] = {
 };
 
 const p_region_t BOOT_RODATA dev_p_regs[] = {
+    { /* .start */ VC_MBOX_PADDR  , /* .end */ VC_MBOX_PADDR   + (1u << PAGE_BITS) },
+    { /* .start */ PM_PADDR       , /* .end */ PM_PADDR        + (1u << PAGE_BITS) },
+    { /* .start */ RNG_PADDR      , /* .end */ RNG_PADDR       + (1u << PAGE_BITS) },
+    { /* .start */ GPIO_PADDR     , /* .end */ GPIO_PADDR      + (1u << PAGE_BITS) },
+    { /* .start */ UART0_PADDR    , /* .end */ UART0_PADDR     + (1u << PAGE_BITS) },
     { /* .start */ SDHC_PADDR     , /* .end */ SDHC_PADDR      + (1u << PAGE_BITS) },
     { /* .start */ USB2_PADDR     , /* .end */ USB2_PADDR      + (1u << PAGE_BITS) },
     { /* .start */ UART_PADDR     , /* .end */ UART_PADDR      + (1u << PAGE_BITS) },
